go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -158,22 +157,18 @@ func routeToEngineRoute(ctx *plancontext.PlanningContext, op *operators.Route) (
 	if err != nil {
 		return nil, err
 	}
-	var vindex vindexes.Vindex
-	var values []evalengine.Expr
-	if op.SelectedVindex() != nil {
-		vindex = op.Selected.FoundVindex
-		values = op.Selected.Values
+
+	ks, _ := ctx.VSchema.DefaultKeyspace() // we don't want to fail here - maybe the op knows where it is going
+
+	rp := &engine.RoutingParameters{
+		Opcode:   op.Routing.OpCode(),
+		Keyspace: ks,
 	}
+	op.Routing.UpdateRoutingParams(ctx, rp)
```

Comment:
It's not clear to me why `UpdateRoutingParams` woud not take care of assigning the `Opcode` and `Keyspace` fields.

Reply:
OpCode is something that all routings need to set, so instead of setting it manually in every `Routing` implementation, I do it on the outside.

The keyspace we are setting here on the outside is the currently selected one, not the one for the query. This is then overwritten in most `Routing`s, but not all.